streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -620,22 +620,35 @@ public final TopologyBuilder copartitionSources(Collection<String> sourceNodes)
      * Returns the copartition groups.
      * A copartition group is a group of source topics that are required to be copartitioned.
      *
+     * @param applicationId
      * @return groups of topic names
      */
-    public Collection<Set<String>> copartitionGroups() {
+    public Collection<Set<String>> copartitionGroups(final String applicationId) {
```

Comment:
Actually, I'm wondering if we could just pass the applicationId when constructing the topology builder? Since this is a public interface, making such changes will affect unit tests, but also potentially other people that are building their own DSL on top of processor API.